core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -524,9 +524,16 @@ class GroupMetadataManager(brokerId: Int,
           val memRecords = fetchDataInfo.records match {
             case records: MemoryRecords => records
             case fileRecords: FileRecords =>
-              buffer.clear()
-              fileRecords.readInto(buffer, 0)
-              MemoryRecords.readableRecords(buffer)
+              val sizeInBytes = fileRecords.sizeInBytes
+              val recordsBuffer = if (sizeInBytes > buffer.capacity) {
+                warn(s"Loaded offsets and group metadata from $topicPartition with buffer larger ($sizeInBytes bytes) than configured offsets.load.buffer.size")
+                ByteBuffer.allocate(sizeInBytes)
```

Comment:
I'm wondering if we should just use the oversized buffer for the remainder of the loading since we have already allocated it. For one thing, that would allow the other buffer to be collected immediately. It may also prevent further allocations if the topic has many oversized messages.

Reply:
Does this mean that we will basically ignore the `offsets.load.buffer.size`, or, rather, that it becomes the "initial" size instead of the limit?

In that case, should we update the docs on the config?

Reply:
The config becomes a soft limit in any case with this patch. I think it does make sense to mention this in the documentation.

Reply:
I've updated the branch with a mean that replaces the buffer in case we need to allocate a bigger one. This was also suggested by @dhruvilshah3 it may be a better idea anyway.
The code does not look as idiomatic as it used to, but at least it "does the job" 🙂I am definitely open to make changes if either of you has a better suggestion!